src/main.cpp

```diff
@@ -658,6 +692,96 @@ class memSafeHeapCallbackHandler
     };
 };
 
+class HookSystemHandler
+{
+private:
+    static void InitializeLua()
+    {
+        // Guaranteed to be loaded before anything vital is called (e.g. AngelReadString)
+        if (gameVersion == MM2_RETAIL)
+        {
+            MM2Lua::Initialize();
+        }
+        else
+        {
+            // no lua support for betas yet
+            MessageBox(NULL, "NOTE: This game version does not currently support Lua scripting.", "MM2Hook", MB_OK | MB_ICONINFORMATION);
+            return;
+        }
+    }
+public:
+    static void Initialize(int argv, char **argc)
+    {
+        // initialize the Lua engine
+        InitializeLua();
+
+        if (gameVersion == MM2_RETAIL)
+        {
+            // hook into the printer
+            $Printer = &PrinterHandler::Print;
+            $PrintString = &PrinterHandler::PrintString;
+
+            /* Won't write to the log file for some reason :(
+            LogFile::Write("Redirecting MM2 output...");
+
+            if (datOutput::OpenLog("mm2.log"))
+            LogFile::Write("Done!\n");
+            else
+            LogFile::Write("FAIL!\n");
+            */
+
+            if (datArgParser::Get("noautodetect"))
```

Comment:
I'm thinking maybe we should change this to an int argument that works like this:

`-autodetect 1` -- Use old behavior
`-autodetect 2` -- Use new behavior (default)